code/game/turfs/turf.dm

```diff
@@ -346,3 +344,9 @@
 				O.singularity_act()
 	ChangeTurf(/turf/space)
 	return(2)
+
+/turf/proc/can_have_cabling()
+	return !density
+
+/turf/proc/can_lay_cable()
+	return can_have_cabling() & !intact
```

Comment:
!intact is for /floor turfs. If they have a tile on them they are 'intact', so you shouldn't be able to lay cables there. Space is never 'intact' so it doesn't interfere with catwalk cables.